contracts/dapis/DapiServer.sol

```diff
@@ -636,119 +636,117 @@ contract DapiServer is
-    /// by the respective Airnodes for the specific bid. The Beacons for which
-    /// the fulfillment data and signature is omitted will be read from
-    /// storage.
-    /// @dev Even though data for Beacons are signed individually, the caller
-    /// is only allowed to use the signatures as a bundle. They cannot omit
-    /// individual signatures or mix-and-match among bundles.
+    /// @notice Updates a data feed that the OEV proxy reads using the
+    /// aggregation signed by the absolute majority of the respective Airnodes
+    /// for the specific bid
+    /// @dev For when the data feed being updated is a Beacon set, an absolute
+    /// majority of Airnodes must sign the resulting value and timestamp. While
+    /// doing so, Airnodes should refer to data signed by the respective
+    /// Airnodes. The absolute majority of this signed data should have similar
+    /// timestamps and values (e.g., within 2 minutes and 1% deviation).
+    /// Freshness should not be required, as OEV proxy data feeds cannot be
+    /// updated using older timestamps.
     /// @param oevProxy OEV proxy that reads the data feed
+    /// @param dataFeedId Data feed ID
     /// @param updateId Update ID
-    /// @param signatureCount Number of signatures in `signedData`
-    /// @param signedData Array of ABI-encoded Airnode address, template ID,
-    /// timestamp, fulfillment data and bid metadata that is signed by the
-    /// respective Airnode for the specific bid
+    /// @param timestamp Signature timestamp
+    /// @param data Update data (an `int256` encoded in contract ABI)
+    /// @param packedOevUpdateSignatures Packed OEV update signatures, which
+    /// include the Airnode address, template ID and these signed with the OEV
+    /// update hash
     function updateOevProxyDataFeedWithSignedData(
         address oevProxy,
+        bytes32 dataFeedId,
         bytes32 updateId,
-        uint256 signatureCount,
-        bytes[] calldata signedData
-    ) external payable override {
-        uint256 beaconCount = signedData.length;
-        bytes32 metadataHash = keccak256(
+        uint256 timestamp,
+        bytes calldata data,
+        bytes[] calldata packedOevUpdateSignatures
+    ) external payable override onlyValidTimestamp(timestamp) {
+        require(
+            timestamp > oevProxyToIdToDataFeed[oevProxy][dataFeedId].timestamp,
+            "Does not update timestamp"
+        );
+        bytes32 oevUpdateHash = keccak256(
             abi.encodePacked(
                 block.chainid,
                 address(this),
-                address(oevProxy),
-                msg.sender,
-                msg.value,
+                oevProxy,
+                dataFeedId,
                 updateId,
-                signatureCount,
-                beaconCount
+                timestamp,
+                data,
+                msg.sender,
+                msg.value
```

Comment:
Since we are using `abi.encodePacked()`, shouldn't we put these 2 before the dynamically sized one (`data`)?

Reply:
The ordering doesn't affect the size of the resulting packing while using `abi.encodePacked()` (it packs/doesn't pad). I think you're thinking about avoiding placing two consecutive dynamically sized variables in a packed encoding because it's ambiguous where one ends and the other one starts, which is not the case here because there is a single dynamically sized variable.